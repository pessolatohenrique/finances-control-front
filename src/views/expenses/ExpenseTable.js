import React from "react";
import moment from "moment";
import {
  TableContainer,
  Table,
  TableHead,
  TableBody,
  TableCell,
  TableRow,
  IconButton,
} from "@mui/material";
import EditIcon from "@material-ui/icons/Edit";
import DeleteIcon from "@material-ui/icons/Delete";
import { StyledTableCell } from "../../constants/CustomStyles";

export function ExpenseTable({ budget, onShowModal, onSetSelectedId }) {
  return (
    <TableContainer>
      <Table>
        <TableHead>
          <TableRow>
            <StyledTableCell sx={{ width: 100 }}>#</StyledTableCell>
            <StyledTableCell>Descrição</StyledTableCell>
            <StyledTableCell>Categoria</StyledTableCell>
            <StyledTableCell>Valor</StyledTableCell>
            <StyledTableCell>Data</StyledTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {budget?.Expenses.map((row) => {
            return (
              <TableRow key={row?.id} data-testid="row-table">
                <TableCell>
                  <IconButton
                    color="primary"
                    aria-label="edit"
                    component="span"
                    onClick={() => window.location.replace(`/gasto/${row?.id}`)}
                  >
                    <EditIcon href={`/gasto/${row?.id}`} />
                  </IconButton>
                  <IconButton
                    color="error"
                    aria-label="delete"
                    component="span"
                    onClick={() => {
                      onSetSelectedId(row?.id);
                      onShowModal();
                    }}
                  >
                    <DeleteIcon />
                  </IconButton>
                </TableCell>
                <TableCell>{row?.Expense?.name}</TableCell>
                <TableCell>
                  {row?.Expense?.userExpenseCategory?.Category?.name}
                </TableCell>
                <TableCell>{row?.value}</TableCell>
                <TableCell>
                  {moment(row?.UserExpense?.transaction_date).format(
                    "DD/MM/YYYY"
                  )}
                </TableCell>
              </TableRow>
            );
          })}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
