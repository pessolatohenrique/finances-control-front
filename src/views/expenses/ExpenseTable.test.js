import moment from "moment";
import { render, screen } from "@testing-library/react";
import { ExpenseTable } from "./ExpenseTable";

const budget = {
  Expenses: [
    {
      id: 1,
      name: "Internet",
      userExpenseCategory: { Category: { name: "Essencial" } },
      UserExpense: {
        value: 300,
        transaction_date: moment().format("YYYY-MM-DD"),
      },
    },
    {
      id: 2,
      name: "Camisetas",
      userExpenseCategory: { Category: { name: "Livre estou" } },
      UserExpense: {
        value: 90,
        transaction_date: moment().format("YYYY-MM-DD"),
      },
    },
  ],
};

describe("Earning Table", () => {
  it("should generate snapshot", () => {
    const { container } = render(<ExpenseTable budget={budget} />);
    expect(container).toMatchSnapshot();
  });

  it("should generate table", () => {
    render(<ExpenseTable budget={budget} />);
    const rowTable = screen.getAllByTestId("row-table");
    expect(rowTable.length).toBe(2);
  });
});
