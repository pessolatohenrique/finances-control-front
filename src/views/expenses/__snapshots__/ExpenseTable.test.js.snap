// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Earning Table should generate snapshot 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Descrição
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Categoria
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Valor
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Data
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          data-testid="row-table"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Internet
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Essencial
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            300
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            17/05/2022
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          data-testid="row-table"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Camisetas
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Livre estou
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            90
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            17/05/2022
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
